### DISCUSSION TITLE
How to properly add new character abilities?

### DISCUSSION THREAD

**Discussion by Aikocho** [TrianglyRU/OrbinautFramework#37] [Q&A] [answered]
Maybe it's something I didn't read in the wiki, or something I didn't find yet, but how would you go about not only adding a character, but their abilities too without problems?

**Comment by TrianglyRU** [comment, marked as answer]
A [guide](https://github.com/TrianglyRU/OrbinautFramework/discussions/31) for adding a new character has already been created by one of the users.

As for adding abilities similar to the existing ones:

- Create a new macro for the action variable.
- Set up a separate script for the action.
- Implement a check for the action within that script. Mid-air abilities are triggered in the scr_player_jump() script. Take a look at the existing actions and how they are called. If you can't find something, use Ctrl+Shift+F for assistance.
- Call the action script in your desired player state (User Event 0 of obj_player)

Orbinaut assumes you have a basic understanding of programming, so I won't go into too much detail. I hope you understand. If you have any further questions, feel free to ask!

**Reply by TrianglyRU** [reply]
I will consider adding several guides about the player to the wiki, but at a later date.

**Reply by Aikocho** [reply]
> I will consider adding several guides about the player to the wiki, but at a later date.

I understand, I'll use the guide above for adding characters, as for the abilities, I can try this too. I just need to look around I guess.

Either way, thank you for this friend.
